Include rule set information on linter rules page (#3228)
dart-lang/site-www: src/tools/linter-rules.md

Use the Dart linter to identify possible problems in your Dart code.
You can use the linter through your IDE
or with the [`dart analyze`](/tools/dart-analyze) command.
For information on how to enable and disable individual linter rules, see
[Enabling individual rules][] in the [analyzer documentation][].

[Enabling individual rules]: guides/language/analysis-options#individual-rules
[analyzer documentation]: /guides/language/analysis-options

This page lists all the linter rules,
with details such as when you might want to use each rule,
what code patterns trigger it, and
how you might fix your code.

## Predefined rule sets

{{site.alert.tip}}
Linter rules (sometimes called _lints_) can have false positives,
and they don’t all agree with each other.
For example, some rules are more appropriate for library packages,
and others are designed for Flutter apps.
{{site.alert.end}}

To avoid the need to individually select compatible linter rules,
**consider starting with one of many pre-packaged linter rule sets:**

[lints]({{site.pub-pkg}}/lints)

> : The official selections of lints that the Dart team encourages using.

Reviewer: Maybe also add: "Dart packages uploaded to pub.dev are always [scored](https://pub.dev/help/scoring) with these lints"

Reviewer 2: Does pub use the core set or the recommended set?

Author: I believe pub scores based on the core set currently.

[flutter_lints]({{site.pub-pkg}}/flutter_lints)

> : The recommended set of lints for Flutter apps, packages, and plugins.

Reviewer: This one should use similar language to the one above, just with "the Flutter team"

> : The recommended set of lints for Flutter apps, packages, and plugins.

Reviewer: Maybe also add: "Flutter packages uploaded to pub.dev are always [scored](https://pub.dev/help/scoring) with these lints"

Reviewer 2: Agree that it's important to say which lints pub.dev uses.

Author: So Flutter packages will be scored with this set rather than `core`?

Reviewer: No, I misspoke. All packages will be scored with 'core'

But flutter_lints includes lints/recommended, which included lints/core.

[effective_dart]({{site.pub-pkg}}/effective_dart)
: A set of lints corresponding to the guidelines in [Effective Dart][].

[pedantic]({{site.pub-pkg}}/pedantic)
: The selective set of lints used internally at Google

>   to enforce best practices.

Reviewer: I would change this to something like: "The set of lints used internally at Google to enforce the Google-internal style for Dart code."

[Enabling individual rules][] in the [analyzer documentation][].

[Enabling individual rules]: guides/language/analysis-options#individual-rules
[analyzer documentation]: /guides/language/analysis-options

This page lists all the linter rules,
with details such as when you might want to use each rule,
what code patterns trigger it, and
how you might fix your code.

## Predefined rule sets

{{site.alert.tip}}
Linter rules (sometimes called _lints_) can have false positives,
and they don’t all agree with each other.
For example, some rules are more appropriate for library packages,
and others are designed for Flutter apps.
{{site.alert.end}}

To avoid the need to individually select compatible linter rules,
consider starting with a linter rule set,
which the following packages provide:

[lints]({{site.pub-pkg}}/lints)
: The official selections of lints that the Dart team encourages using.

[flutter_lints]({{site.pub-pkg}}/flutter_lints)
: The recommended set of lints for Flutter apps, packages, and plugins.

[effective_dart]({{site.pub-pkg}}/effective_dart)
: A set of lints corresponding to the guidelines in [Effective Dart][].

[pedantic]({{site.pub-pkg}}/pedantic)

> : The selective set of lints used internally at Google
>   to enforce best practices.

Reviewer 2: How about this, @mit-mit: 

```suggestion
: The set of rules used for all Google-internal Dart code.
```

> To learn how to utilize a specific rule set,

Reviewer 2: ```suggestion
To learn how to use a specific rule set,
```

see [Enabling and disabling linter rules][].

[Enabling and disabling linter rules]: /guides/language/analysis-options#enabling-linter-rules
[Effective Dart]: /guides/language/effective-dart

> ## Rule types

Reviewer 2: Good idea to add these section heads

Author: Yeah...they were starting to blend together without them!
